### PR TITLE
WIP - feat: add regional support

### PR DESCRIPTION
**Pull request by thinkingserious**
Adds Twilio regional support.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: src/main/java/com/twilio/Twilio.java [hunk 1]
```diff
@@ -121,6 +155,24 @@ public static void setAccountSid(final String accountSid) {
         Twilio.accountSid = accountSid;
     }
 
+    /**
+     * Set the region.
+     *
+     * @param region region to make request
+     */
+    public static void setRegion(final String region) {
+        Twilio.region = region;
```
Comment (childish-sambino) [inline src/main/java/com/twilio/Twilio.java:164]: Changing the region should also invalidate the client like above.

File: src/main/java/com/twilio/Twilio.java [hunk 2]
```diff
@@ -67,6 +69,38 @@ public static void init(final String username, final String password, final Stri
...
+    /**
+     * Initialize the Twilio environment.
+     *
+     * @param username account to use
+     * @param password auth token for the account
+     * @param accountSid account sid to use
+     * @param region region to make request
+     * @param edge edge to make request
+     */
+    public static void init(final String username, final String password, final String accountSid, final String region, final String edge) {
```
Comment (childish-sambino) [inline src/main/java/com/twilio/Twilio.java:96]: Prefer not to add these 2. Too much overloading. For instance, you can't have just region w/o edge because it would conflict with another constructor (unless you explicitly pass a null edge).

File: src/test/java/com/twilio/TwilioTest.java [hunk 3]
```diff
@@ -135,4 +137,97 @@ public void testValidateSslCertificateSuccess() {
...
+            networkHttpClient.makeRequest(request);
+            times = 1;
+            result = new Response("", 200);
+        }};
+
+        Twilio.init("accountSid", "authToken");
+        Twilio.setEdge("edge");
+        Request request = new Request(HttpMethod.GET, "api", "/test", null, Twilio.getEdge());
+        //Request request = new Request(HttpMethod.GET, "https://api.edge.us1.twilio.com/test");
+        Response result = networkHttpClient.makeRequest(request);
```
Comment (eshanholtz) [inline src/test/java/com/twilio/TwilioTest.java:154]: Why are you use `networkHttpClient` here instead of `TwilioRestClient client = Twilio.getRestClient();` like in [this regional test](https://github.com/twilio/twilio-java/pull/525/files#diff-019d944bb76bdd4be40cc700e5217442R159)?

File: src/test/java/com/twilio/TwilioTest.java [hunk 4]
```diff
@@ -18,7 +17,7 @@
 import java.util.concurrent.Executors;
 
 import mockit.Mocked;
-import mockit.NonStrictExpectations;
+import mockit.Expectations;
```
Comment (eshanholtz) [inline src/test/java/com/twilio/TwilioTest.java:20]: Why this change?

Comment (thinkingserious) [reply src/test/java/com/twilio/TwilioTest.java:20]: `NonStrictExpectations` is deprecated.

File: src/test/java/com/twilio/TwilioTest.java (hunk 3 above)
Comment (thinkingserious) [reply src/test/java/com/twilio/TwilioTest.java:154]: Reverted back, no luck.
